python_modules/libraries/dagster-duckdb/dagster_duckdb/io_manager.py

```diff
@@ -279,13 +280,16 @@ def get_select_statement(table_slice: TableSlice) -> str:
 
     @staticmethod
     @contextmanager
-    def connect(context, _):
+    def connect(context, _, **kwargs):
```

Comment:
did you consider determining whether to do a read connection by running `isinstance(context, InputContext)` in this function rather than doing the `kwargs` method?

Reply:
I used kwargs because you had mentionned it in the issue:

```
We can make this a kwarg with defaults so that any custom DBClients out there won't see breaking changes.
```

But I don't have a strong opinion about it. I mainly do software engineering with C#, so I don't have as much experience with the best practices in python yet + I am not certain if this follows your conventions elsewhere.

Also please note I was not able to test this code since I am in windows and it seemed quite complicated to setup.